src/App.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  componentDidMount() {
```

Comment:
I'm not sure, but as I understand it, if we start actions in componentDidMount, we must stop them in componentWillUnmount.
You start in componentDidMount
   document.addEventListener('contextmenu'......)
   document.addEventListener('click',.....)
   window.setInterval(()....)
So you need to stop in componentWillUnmount:
       document.removeEventListener('click', ....);
     document.removeEventListener('contextmenu', ....);
     window.clearInterval(.......);

Reply:
Thanks! Fixed it

> I'm not sure, but as I understand it, if we start actions in componentDidMount, we must stop them in componentWillUnmount. You start in componentDidMount document.addEventListener('contextmenu'......) document.addEventListener('click',.....) window.setInterval(()....) So you need to stop in componentWillUnmount: document.removeEventListener('click', ....); document.removeEventListener('contextmenu', ....); window.clearInterval(.......);